test/e2e/atlas/search_nodes_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e || (atlas && search_nodes)
+
+package atlas_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-cli/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+const minSearchNodesMDBVersion = "6.0"
+
+func TestSearchNodes(t *testing.T) {
+	cliPath, err := e2e.AtlasCLIBin()
+	req := require.New(t)
+	req.NoError(err)
+
+	g := newAtlasE2ETestGenerator(t)
+	g.generateProject("searchNodes")
+	g.tier = tierM10
+	g.mDBVer = minSearchNodesMDBVersion
+	g.generateCluster()
+
+	t.Run("Verify no search node setup yet", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			clustersEntity,
+			searchEntity,
+			nodesEntity,
+			"list",
+			"--clusterName", g.clusterName,
+			"--projectId", g.projectID,
+			"-o=json",
+		)
+
+		require.Error(t, cmd.Run())
+	})
+
+	t.Run("Create search node", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			clustersEntity,
+			searchEntity,
+			nodesEntity,
+			"create",
+			"--clusterName", g.clusterName,
+			"--projectId", g.projectID,
+			"--file", "search_nodes_spec.json",
+			"-w",
+			"-o=json",
+		)
+
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		resp = bytes.TrimLeft(resp, ".")
+
+		require.NoError(t, err, resp)
+		var searchNode atlasv2.ApiSearchDeploymentResponse
+		require.NoError(t, json.Unmarshal(resp, &searchNode))
+
+		assert.Equal(t, *searchNode.GroupId, g.projectID)
```

Comment:
expected, got and let's use getters to avoid pointer logic
```suggestion
		assert.Equal(t, g.projectID, searchNode.GetGroupId())
```

Reply:
Fixed